### ISSUE TITLE
Changes.hasNext() never return false when last_seq row is received 

### ISSUE THREAD

**Issue by atyyeung** [cloudant/java-cloudant#390]
Please include the following information in your ticket.

- CloudantClient (java-cloudant) version(s) that are affected by this issue.
2.90

- Java version (including vendor and platform).
java version "1.8.0_131"
Java(TM) 2 Runtime Environment, Standard Edition (IBM build 1.8.0_131-b11 24_Apr_2017_14_30 Mac OS X x64(SR4 FP5))

- If you're using the optional okhttp dependency.
No

- A small code sample that demonstrates the issue.
```
while (changeFeed.hasNext()) {
    // process change feed
}
```
The feed is stopped but we could never get out of the while loop because changeFeed.hasNext() never returns false.  

`Changes.readNextRow()` does not handle the case when `row.startsWith("{\"last_seq\":")` is true.  It never break the `while(true)` loop when it receives an end of feed signal from cloudant.

**Comment by ricellis** [comment]
@atyyeung this fix is available in [2.11.0](https://github.com/cloudant/java-cloudant/releases/tag/2.11.0)

**Comment by atyyeung** [comment]
Great, thank you.
